Sources/MongoSwift/BSON/AnyBSONValue.swift

```diff
@@ -2,25 +2,49 @@ import Foundation
 
 /// A struct wrapping a `BSONValue` type that allows for encoding/
 /// decoding `BSONValue`s of unknown type.
+
 public struct AnyBSONValue: Codable, Equatable, Hashable {
-    // TODO: conform all `BSONValue` types to `Hashable` (SWIFT-320).
+    // swiftlint:disable:next cyclomatic_complexity
     public func hash(into hasher: inout Hasher) {
         hasher.combine(self.value.bsonType)
-        // A few types need to be handled specifically because their string representations aren't sufficient or
-        // performant.
-        if let date = self.value as? Date {
-            // `Date`'s string conversion omits milliseconds and smaller time units, and using a string formatter is
-            // expensive. Instead, we just include the time interval itself.
-            hasher.combine(date.timeIntervalSince1970)
-        } else if let binary = self.value as? Binary {
-            // `Binary`'s string representation omits the data itself, so we include its hashValue.
-            hasher.combine(binary.data)
-            hasher.combine(binary.subtype)
-        } else if let arr = self.value as? [BSONValue] {
-            // To factor in every item in the array, we include the arrays extended JSON representation.
-            hasher.combine((["value": arr] as Document).extendedJSON)
+
+        switch self.value {
```

Comment:
I think you are missing `BSONNull`, `Timestamp`, `BSONUndefined`, and `DBPointer` here.

Reply:
Now that I look at the rest of this file, I see we are missing the last 3 of those from the `init(from decoder: Decoder)` method below where we try all of the different BSON types. So I'm guessing that's why you missed them here since I told you to look at that for an example... That is a bad bug!! 😩 
I opened a JIRA ticket about this: [SWIFT-508](https://jira.mongodb.org/browse/SWIFT-508)